### PR TITLE
Add aks envs

### PR DESCRIPTION
**Pull request by RMcVelia**
### Context

Add config files for aks environments

https://trello.com/c/jyheDesS/177-register-replicate-each-environment-to-aks

### Changes proposed in this pull request

- Add terraform and rails config files for each aks env
- update Makefile
- updates to workflows

### Guidance to review

make env deploy-plan

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Comment by saliceti** [comment]
[Register - Replicate each environment to AKS](https://trello.com/c/jyheDesS/177-register-replicate-each-environment-to-aks)

File: Makefile [hunk 1]
```diff
@@ -125,10 +119,36 @@ dv_review_aks: ## make dv_review_aks deploy APP_NAME=2222 CLUSTER=cluster1
...
 	echo https://register-$(APP_NAME).$(CLUSTER).development.teacherservices.cloud will be created in aks
-#	$(eval APP_NAME_SUFFIX=dv-review-$(APP_NAME))
-#	$(eval backend_key=-backend-config=key=pr-$(APP_NAME).tfstate)
-#	$(eval export TF_VAR_app_name_suffix=pr-$(APP_NAME))
 
+qa_aks:
+	$(eval include global_config/qa_aks.sh)
+	$(eval DEPLOY_ENV=qa_aks)
+	$(eval DTTP_HOSTNAME=traineeteacherportal-dv)
+	$(eval paas_env=qa)
```
Comment (smneal) [inline Makefile:127]: Should this be here? Should there be one in staging as well if so?

File: Makefile [hunk 2]
```diff
@@ -125,10 +119,36 @@ dv_review_aks: ## make dv_review_aks deploy APP_NAME=2222 CLUSTER=cluster1
...
+	$(eval DTTP_HOSTNAME=traineeteacherportal-pp)
+	$(eval export TF_VARS=-var config_short=${CONFIG_SHORT} -var service_short=${SERVICE_SHORT} -var service_name=${SERVICE_NAME} -var azure_resource_prefix=${RESOURCE_NAME_PREFIX})
+
+production_aks:
+	$(eval include global_config/production_aks.sh)
+	$(if $(CONFIRM_PRODUCTION), , $(error Can only run with CONFIRM_PRODUCTION))
+	$(eval DEPLOY_ENV=production_aks)
+	$(eval HOST_NAME=www)
+	$(eval DTTP_HOSTNAME=traineeteacherportal)
+	$(eval paas_env=production)
```
Comment (smneal) [inline Makefile:143]: See comment about qa paas_env.

File: Makefile (hunk 1 above)
Comment (RMcVelia) [reply Makefile]: removed as not required for aks env

File: Makefile (hunk 2 above)
Comment (RMcVelia) [reply Makefile]: removed as not required for aks env

**Review by smneal — APPROVED** [review]
LGTM

File: config/environments/production_aks.rb [hunk 3]
```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
```
Comment (saliceti) [inline config/environments/production_aks.rb:1]: As discussed, all environments should call production.rb

**Review by saliceti — APPROVED** [review]

**Pull request merged**